spring-pulsar/src/test/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainerTests.java

```diff
@@ -202,4 +204,60 @@ void negativeAckRedeliveryBackoff() throws Exception {
 		pulsarClient.close();
 	}
 
+	@Test
+	void deadLetterPolicy() throws Exception {
+		Map<String, Object> config = new HashMap<>();
+		config.put("topicNames", Collections.singleton("dpmlct-016"));
+		config.put("subscriptionName", "dpmlct-sb-016");
+		config.put("ackTimeoutMillis", 1);
+		DeadLetterPolicy deadLetterPolicy = DeadLetterPolicy.builder().maxRedeliverCount(1).deadLetterTopic("dlq-topic")
+				.build();
+		config.put("deadLetterPolicy", deadLetterPolicy);
+
+		final PulsarClient pulsarClient = PulsarClient.builder().serviceUrl(getPulsarBrokerUrl()).build();
+		final DefaultPulsarConsumerFactory<String> pulsarConsumerFactory = new DefaultPulsarConsumerFactory<>(
+				pulsarClient, config);
+
+		CountDownLatch dlqLatch = new CountDownLatch(1);
+		CountDownLatch latch = new CountDownLatch(6);
+
+		PulsarContainerProperties dlqContainerProperties = new PulsarContainerProperties();
+		dlqContainerProperties
+				.setMessageListener((PulsarRecordMessageListener<?>) (consumer, msg) -> dlqLatch.countDown());
+		dlqContainerProperties.setSchema(Schema.STRING);
+		dlqContainerProperties.setSubscriptionType(SubscriptionType.Shared);
+		dlqContainerProperties.setTopics(new String[] { "dlq-topic" });
+		DefaultPulsarMessageListenerContainer<String> dlqContainer = new DefaultPulsarMessageListenerContainer<>(
+				pulsarConsumerFactory, dlqContainerProperties);
+		dlqContainer.start();
+
+		PulsarContainerProperties pulsarContainerProperties = new PulsarContainerProperties();
+		pulsarContainerProperties.setMessageListener((PulsarRecordMessageListener<?>) (consumer, msg) -> {
+			latch.countDown();
+			if (((String) msg.getValue()).endsWith("4")) {
+				throw new RuntimeException("fail");
+			}
+		});
+		pulsarContainerProperties.setSchema(Schema.STRING);
+		pulsarContainerProperties.setSubscriptionType(SubscriptionType.Shared);
+		DefaultPulsarMessageListenerContainer<String> container = new DefaultPulsarMessageListenerContainer<>(
+				pulsarConsumerFactory, pulsarContainerProperties);
+		container.start();
+
+		Map<String, Object> prodConfig = Collections.singletonMap("topicName", "dpmlct-016");
+		final DefaultPulsarProducerFactory<String> pulsarProducerFactory = new DefaultPulsarProducerFactory<>(
+				pulsarClient, prodConfig);
+		final PulsarTemplate<String> pulsarTemplate = new PulsarTemplate<>(pulsarProducerFactory);
+		for (int i = 0; i < 5; i++) {
+			pulsarTemplate.send("hello john doe" + i);
+		}
+
+		// DLQ consumer should receive 1 msg
+		assertThat(dlqLatch.await(10, TimeUnit.SECONDS)).isTrue();
+		// Normal consumer should receive 5 msg + 1 re-delivery
+		assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
+		container.stop();
```

Comment:
These may not always be closed. The other tests in here have the same issue though. I am fine if we clean that up in a follow on PR or commit.